test/unit-tests/sys-info-base.ts

```diff
@@ -214,12 +210,33 @@ describe("sysInfoBase", () => {
 			});
 		});
 
+		describe("getJavaCompilerVersion", () => {
+			const verifyJavaCompilerVersion = async (javaCompilerVersion: string, resultStream: string): Promise<void> => {
+				const javaCompilerChildProcessResult: any = {
+					[resultStream]: `javac ${javaCompilerVersion}`
+				};
+
+				javaCompilerChildProcessResult.javacVersion = { result: javaCompilerChildProcessResult };
+				testInjector = createTestInjector(javaCompilerChildProcessResult, { isWindows: false, isDarwin: true }, null);
+				sysInfoBase = testInjector.resolve("sysInfoBase");
+				const actualJavaCompilerVersion = await sysInfoBase.getJavaCompilerVersion();
+				assert.deepEqual(actualJavaCompilerVersion, javaCompilerVersion);
+			};
+
+			it("returns correct javac version when it is printed on stderr (Java 8)", () => {
+				return verifyJavaCompilerVersion("1.8.0_152", "stderr");
+			});
+
+			it("returns correct javac version when it is printed on stdout (Java 9)", () => {
+				return verifyJavaCompilerVersion("9.0.1", "stdout");
```

Comment:
@rosen-vladimirov some distributions of java return `9` when `java -version` is called, in which case I think the regex might break. I haven't tested it but I encountered such an issue in the past.

Reply:
Thanks for pointing this. In case 9 is returned, the current code will work fine, but [this one](https://github.com/NativeScript/nativescript-cli/pull/3294/files#diff-8288e0e434a7344884ce096d3dfd239bR115) will fail. I'll handle the case there.

Reply:
@Plamen5kov fixed in this PR https://github.com/NativeScript/nativescript-cli/pull/3294/

Reply:
@rosen-vladimirov I tested it, and CLI provides a message to install java 1.8.0.